native/src/main/java/io/ballerina/lib/avro/Utils.java

```diff
@@ -31,11 +35,25 @@ private Utils() {
 
     public static final String AVRO_SCHEMA = "avroSchema";
     public static final String ERROR_TYPE = "Error";
-    public static final String JSON_PROCESSING_ERROR = "JSON processing error";
-    public static final String DESERIALIZATION_ERROR = "Unable to deserialize the byte value";
+    public static final String SERIALIZATION_ERROR = "Avro serialization error";
+    public static final String DESERIALIZATION_ERROR = "Avro deserialization error";
 
     public static BError createError(String message, Throwable throwable) {
         BError cause = ErrorCreator.createError(throwable);
         return ErrorCreator.createError(getModule(), ERROR_TYPE, StringUtils.fromString(message), cause, null);
     }
+
+    public static Type getMutableType(Type dataType) {
+        if (dataType.getTag() != TypeTags.INTERSECTION_TAG) {
+            return dataType;
+        }
+        IntersectionType intersectionType = (IntersectionType) dataType;
+        for (Type type : intersectionType.getConstituentTypes()) {
+            Type referredType = TypeUtils.getImpliedType(type);
+            if (TypeUtils.getImpliedType(intersectionType.getEffectiveType()).getTag() == referredType.getTag()) {
+                return referredType;
+            }
+        }
+        throw new IllegalStateException("Unsupported intersection type found.");
+    }
```

Comment:
My original idea was that can't we simplify this like the below?

```suggestion
        return TypeUtils.getImpliedType(type);
    }
```